### PR TITLE
Bug patching new elements

### PR DESCRIPTION
**Pull request by Aluok**
This is a solution to issue #4. No features added. 

I assumed that the tests with tampered fixture meant that you wanted to keep any functions that are defined, even getters/setters. Is that right?
The way it's currently done might need some input to be improved (in a sense, I don't really like the way it's done, but I don't see another...

Thanks for the project.

### CODE REVIEW

File: src/Phpro/SoapClient/CodeGenerator/Patcher.php [hunk 1]
```diff
@@ -38,29 +38,58 @@ public function patch($original, $newContent)
...
+        $this->filesystem->addToFile($original, $functions, $endClassPos);
+        $this->cleanTmpFiles([$tmpFile, $patchFile, $tmpFilePatched]);
+    }
+
+    /**
+     * @param string $original
+     *
+     * @return string
+     */
+    protected function getFunctions($original)
```
Comment (veewee) [inline src/Phpro/SoapClient/CodeGenerator/Patcher.php]: This method might need some tweeking: what about private, protected and functions without the public keyword?
Does it also work for methods without docblocks?
This functionality definetly needs some tests.

File: src/Phpro/SoapClient/Util/Filesystem.php [hunk 2]
```diff
@@ -78,4 +78,22 @@ public function replaceFile($file, $new, $backup = true)
 
         $this->putFileContents($file, $this->getFileContents($new));
     }
+
+    /**
+     * @param string $file
+     * @param string $newContent
+     * @param int $pos the position at which to insert the content. (default to NULL, understood as append)
+     */
+    public function addToFile($file, $newContent, $pos = null)
```
Comment (veewee) [inline src/Phpro/SoapClient/Util/Filesystem.php]: The $pos is added by the implementation of this method, shouldn't it be a required parameter?

**Comment by veewee** [comment]
Hi @Aluok,

The tests with tampered fixtures indeed meant that I wanted to keep any function that is not available in the new class. The downside of this is that it won't be able to remove properties that don't exist in the WSDL anymore.

The code looks pretty good but doesn't cover every possible case. I've added some small remarks to the code. We definitely need some more unit tests to make sure every works as expected.

In a next phase, I am planning to add [zend-code](http://framework.zend.com/manual/current/en/modules/zend.code.generator.introduction.html) instead of current patch implementation. This will make it possible to write your own code generation rules to speed up your development process even more.
Example rules could be: If a WSDL object name ends with `Request` -> Implement the `RequestInterface`. 

Anyway, this sure is a wanted feature. Thanks for your work! :)

**Comment by veewee** [comment]
Hello @Aluok,

Since the new code generation commands are almost ready, I don't think it is a good idea to put any more time in this PR.
Feel free to check out #10 and give your remarks.

Thanks for the PR anyway!

**Comment by Aluok** [comment]
Hello @veewee,

No problem. I didn't had much time, for the last days, to work more on it anyway. I'll try to give a look at #10 as soon as I can.
Thank you for your work on this project.

**Pull request closed without merging**